src/payment/strategies/googlepay/googlepay-payment-processor.ts

```diff
@@ -19,25 +22,28 @@ import {
     GooglePayAddress,
     GooglePayClient,
     GooglePayInitializer,
-    GooglePayPaymentDataRequestV1,
+    GooglePayPaymentDataRequestV2,
+    GooglePayScriptLoader,
     GooglePaySDK,
-    TokenizePayload
-} from './googlepay';
-import GooglePayScriptLoader from './googlepay-script-loader';
+    TokenizePayload,
+} from './';
 
 export default class GooglePayPaymentProcessor {
-    private _googlePaymentsClient?: GooglePayClient;
+    private _googlePayClient?: GooglePayClient;
     private _methodId?: string;
-    private _googlePaymentDataRequest?: GooglePayPaymentDataRequestV1;
+    private _paymentDataRequest?: GooglePayPaymentDataRequestV2;
+    private _shippingStrategyActionCreator: ShippingStrategyActionCreator;
 
     constructor(
         private _store: CheckoutStore,
         private _paymentMethodActionCreator: PaymentMethodActionCreator,
         private _googlePayScriptLoader: GooglePayScriptLoader,
         private _googlePayInitializer: GooglePayInitializer,
-        private _requestSender: RequestSender,
-        private _billingAddressActionCreator: BillingAddressActionCreator
-    ) { }
+        private _billingAddressActionCreator: BillingAddressActionCreator,
+        private _requestSender: RequestSender
+    ) {
+        this._shippingStrategyActionCreator = new ShippingStrategyActionCreator(createShippingStrategyRegistry(this._store, this._requestSender));
```

Comment:
I'm slightly confused here. What are you trying to do? First all, all instances that could induce side effects should be injected. Secondly, you shouldn't need to inject `ShippingStrategyActionCreator` - just inject `ConsignmentActionCreator` if you need to update shipping addresses.